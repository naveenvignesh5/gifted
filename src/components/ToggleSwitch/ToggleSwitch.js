import React from 'react';

import './ToggleSwitch.sass';

const ToggleSwitch = (props) => (
    <label className="switch">
        <input {...props} type="checkbox"/>
        <span className="slider round"/>
    </label>
);

export default ToggleSwitch;
