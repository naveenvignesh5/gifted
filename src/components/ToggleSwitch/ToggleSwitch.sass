@import '../../styles/theme'
    
.switch
    position: relative
    display: inline-block
    width: 3.75rem
    height: 2.125rem

    input
        opacity: 0
        width: 0
        height: 0

.slider
    position: absolute
    cursor: pointer
    top: 0
    left: 0
    right: 0
    bottom: 0
    background-color: #ccc
    -webkit-transition: .4s
    transition: .4s

    &:before
        position: absolute
        content: ""
        height: 1.625rem
        width: 1.625rem
        left: 0.25rem
        bottom: 0.25rem
        background-color: white
        -webkit-transition: .4s
        transition: .4s

input
    &:checked + .slider
        background-color: $primary

    &:focus + .slider
        box-shadow: 0 0 0.0625rem $primary

    &:checked + .slider:before
        -webkit-transform: translateX(1.625rem)
        -ms-transform: translateX(1.625rem)
    transform: translateX(1.625rem)

.slider.round
    border-radius: 2.125rem

    &:before
        border-radius: 50%