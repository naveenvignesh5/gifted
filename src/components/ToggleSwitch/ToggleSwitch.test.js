import React from "react";
import renderer from "react-test-renderer";

import ToggleSwitch from "./ToggleSwitch";

describe("ToggleSwitch", () => {
  it("snapshot renders", () => {
    const component = renderer.create(
      <ToggleSwitch />
    );
    let tree = component.toJSON();
    expect(tree).toMatchSnapshot();
  });
});
