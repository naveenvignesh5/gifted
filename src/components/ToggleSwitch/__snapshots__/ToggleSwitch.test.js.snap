// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ToggleSwitch snapshot renders 1`] = `
<label
  className="switch"
>
  <input
    type="checkbox"
  />
  <span
    className="slider round"
  />
</label>
`;
